File: perf_num.js
var num = 500



for(var i=1;i<=num;i++)
{
    var sum =0
    a = i/2
    for(var j=1;j<= a;j++)
    {
        if(i%j == 0)
        {
            sum+=j
        }

    if(i == sum)
        console.log("perfect number: "+i)
    }
}

File: perf_num.py
print("enter the value for n: ")
num = int(input())

for i in range(0,num):
    sum = 0
    a = int(i/2)
    for j in range(1,a):
        if(i % j == 0):
            sum=sum+j
        j+=1
    if(sum==i):
        print("perfect number: ", i)